### PR TITLE
Bug 1856502: Added Elasticsearch proxy resources

### PR DESCRIPTION
**Pull request by vimalk78**
fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1856502

### CODE REVIEW

File: pkg/apis/logging/v1/clusterlogging_types.go [hunk 1]
```diff
@@ -150,6 +150,12 @@ type ElasticsearchSpec struct {
 
 	// +optional
 	RedundancyPolicy elasticsearch.RedundancyPolicyType `json:"redundancyPolicy"`
+
+	// The resource requirements for ElasticsearchProxy
+	//
+	// +nullable
+	// +optional
+	ProxyResources *v1.ResourceRequirements `json:"proxyResources"`
```
Comment (periklis) [inline pkg/apis/logging/v1/clusterlogging_types.go:158]: To keep following the pattern (e.g. See `KibanaSpec.ProxySpec`), splitting to a separate `ProxySpec` struct is more appropriate. Grouping proxy configuration in one new struct enables backward compatible additions. (cc @ewolinetz )

Comment (vimalk78) [reply pkg/apis/logging/v1/clusterlogging_types.go]: updated

File: pkg/apis/logging/v1/clusterlogging_types.go [hunk 2]
```diff
@@ -150,6 +150,9 @@ type ElasticsearchSpec struct {
 
 	// +optional
 	RedundancyPolicy elasticsearch.RedundancyPolicyType `json:"redundancyPolicy"`
+
+	// Specification of the Kibana Proxy component
+	ProxySpec `json:"proxy,omitempty"`
```
Comment (periklis) [inline pkg/apis/logging/v1/clusterlogging_types.go:155]: This reuses unfortunately the same proxy spec from kibana. The proxy component refers for kibana to `oauth-proxy`, but for elasticsearch to `elasticsearch-proxy`. The two ProxySpecs should be different structs. In addition please align the comment block for this field to match the purpose.

File: manifests/4.6/logging.openshift.io_clusterloggings_crd.yaml [hunk 3]
```diff
@@ -245,6 +245,41 @@ spec:
                       description: Define which Nodes the Pods are scheduled on.
                       nullable: true
                       type: object
+                    proxy:
+                      description: Specification of the Kibana Proxy component
```
Comment (ewolinetz) [inline manifests/4.6/logging.openshift.io_clusterloggings_crd.yaml:249]: s/Kibana/Elasticsearch/

File: pkg/apis/logging/v1/clusterlogging_types.go [hunk 4]
```diff
@@ -150,6 +150,9 @@ type ElasticsearchSpec struct {
 
 	// +optional
 	RedundancyPolicy elasticsearch.RedundancyPolicyType `json:"redundancyPolicy"`
+
+	// Specification of the Kibana Proxy component
```
Comment (ewolinetz) [inline pkg/apis/logging/v1/clusterlogging_types.go:154]: s/Kibana/Elasticsearch/

File: pkg/apis/logging/v1/clusterlogging_types.go (hunk 2 above)
Comment (ewolinetz) [reply pkg/apis/logging/v1/clusterlogging_types.go:155]: i disagree, i think it is acceptable to reuse this spec given its just:
```
type ProxySpec struct {
	// +nullable
	Resources *v1.ResourceRequirements `json:"resources"`
}
```

i think it is unnecessary to create another struct that represents this same thing but just for elasticsearch's proxy

**Comment by ewolinetz** [comment]
please address the comment referencing kibana otherwise :+1:

File: pkg/apis/logging/v1/clusterlogging_types.go (hunk 2 above)
Comment (periklis) [reply pkg/apis/logging/v1/clusterlogging_types.go:155]: What if we want to extend this struct with kibana's oauth-proxy only fields?

Comment (ewolinetz) [reply pkg/apis/logging/v1/clusterlogging_types.go:155]: we can add fields specific to oauth at that time

Comment (periklis) [reply pkg/apis/logging/v1/clusterlogging_types.go:155]: But the struct will be a union of fields for oauth-proxy and fields for both proxies. Isn't this awkward?

Comment (ewolinetz) [reply pkg/apis/logging/v1/clusterlogging_types.go]: i dont think so, its for a proxy... it's delineated by function. similar to logstore or collection or visualization or forwarder

**Pull request merged**